### PR TITLE
Remove unnecessary final on params and local vars

### PR DESCRIPTION
**Pull request by trask**
This is not the style we follow (anymore).

Also thanks to effectively final in Java 8.

### CODE REVIEW

File: buildSrc/src/main/java/io/opentelemetry/instrumentation/gradle/muzzle/MuzzlePlugin.java [hunk 1]
```diff
@@ -212,7 +212,7 @@ private static TaskProvider addMuzzleTask(
       Artifact versionArtifact,
       Project instrumentationProject,
       TaskProvider<?> runAfter) {
-    final String taskName;
```
Comment (anuraaga) [inline buildSrc/src/main/java/io/opentelemetry/instrumentation/gradle/muzzle/MuzzlePlugin.java:215]: Still recommend `final` when variable isn't initialized inline

Comment (anuraaga) [reply buildSrc/src/main/java/io/opentelemetry/instrumentation/gradle/muzzle/MuzzlePlugin.java]: Applies to many of the changes in this PR :P

File: instrumentation/jms-1.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jms/MessagePropertyGetter.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
   @Override
   public String get(MessageWithDestination carrier, String key) {
     String propName = key.replace("-", MessagePropertySetter.DASH);
-    Object value;
+    final Object value;
```
Comment (anuraaga) [inline instrumentation/jms-1.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jms/MessagePropertyGetter.java:26]: :+1:

**Review by anuraaga — APPROVED** [review]

File: buildSrc/src/main/java/io/opentelemetry/instrumentation/gradle/muzzle/MuzzlePlugin.java (hunk 1 above)
Comment (trask) [reply buildSrc/src/main/java/io/opentelemetry/instrumentation/gradle/muzzle/MuzzlePlugin.java]: oh yes, reverted them all, though I'm not sure I understand the purpose for marking these vars final?

Comment (anuraaga) [reply buildSrc/src/main/java/io/opentelemetry/instrumentation/gradle/muzzle/MuzzlePlugin.java]: When initialization happens deep in some control structure, accidentally double-initializing is a relatively common bug (or at least that's what I was taught when learning Java a long time ago :P)

**Review by iNikem — APPROVED** [review]

**Pull request merged**